src/server/services/interfaces/AnalyticsLoggerService.ts

```diff
@@ -0,0 +1,10 @@
+export interface AnalyticsLoggerService<T> {
+  /**
+   * @param {T} pageViewHit - A page view request payload.
+   */
+  logRedirectAnalytics: (pageViewHit: T) => void
+
+  generateCookie: (
+    cookie?: string,
+  ) => [string, string, { maxAge: number }] | null
```

Comment:
Should we start to give a name to the type `[string, string, { maxAge: number }] | null`?

Reply:
It's an array - basically arguments to be spread onto `res.cookie`. We could refactor this later